edns.go

```diff
@@ -125,7 +125,11 @@ func (rr *OPT) UDPSize() uint16 {
 
 // SetUDPSize sets the UDP buffer size.
 func (rr *OPT) SetUDPSize(size uint16) {
-	rr.Hdr.Class = size
+	switch {
```

Comment:
why do we need this? size > MaxSize doesn't make much sense either because in the input in this function is an uint16. If someone wants to set a smaller limit we should allow it

Reply:
Values < 512 will be ignored by the server.  Values > Max wll be ignored if we're the client.  I've removed this as the check in client are what I was focussed on.

Reply:
MaxMsgSize might be set smaller on a platform where there is less message, so checking MaxMsgSize makes sense in general (though it can't be exceeded with the default MaxMsgSize).